wdl-lint/src/rules/redundant_assignment.rs

```diff
@@ -0,0 +1,120 @@
+//! A lint rule for flagging redundant `= None` assignments for
+//! optional inputs.
+
+use wdl_analysis::Diagnostics;
+use wdl_analysis::VisitReason;
+use wdl_analysis::Visitor;
+use wdl_ast::AstNode;
+use wdl_ast::AstToken;
+use wdl_ast::Diagnostic;
+use wdl_ast::Span;
+use wdl_ast::SyntaxElement;
+use wdl_ast::SyntaxKind;
+use wdl_ast::v1::BoundDecl;
+use wdl_ast::v1::Expr;
+use wdl_ast::v1::LiteralExpr;
+
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the redundant assignment rule.
+const ID: &str = "RedundantAssignment";
+
+/// Create a "redundant `= None` assignment" diagnostic
+fn redundant_none(span: Span, name: &str) -> Diagnostic {
+    Diagnostic::note(format!(
+        "redundant assignment of `None` to optional input `{name}`"
+    ))
+    .with_rule(ID)
+    .with_highlight(span)
+    .with_fix(format!("remove `= None` for input `{name}`"))
+}
+
+/// A rule that identifies redundant `= None` assignments for
+/// optional inputs.
+#[derive(Debug, Default, Clone)]
+pub struct RedundantAssignment;
+
+impl Rule for RedundantAssignment {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Flags redundant explicit assignment of `None` to optional inputs."
+    }
+
+    fn explanation(&self) -> &'static str {
+        "The specification states that an optional input declaration (e.g., `String? foo`) is \
+         implicitly initialized to `None` if no other value or default is provided. Therefore \
+         explicitly writing `String? foo = None` is equivalent to `String? foo` but adds \
+         unnecessary verbosity. This rule encourages omitting the redundant `= None` for optional \
+         inputs.
+
+         For example, 
+        `String? my_optional_input = None` should be simplified to `String? my_optional_input`."
```

Comment:
```suggestion
         unnecessary verbosity."
```
Nit: IMO the text I deleted here isn't adding much value. I'd prefer to keep this text shorter.